File: javascript/easy/largestNumberAtLeastTwiceOfOthers.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var dominantIndex = function(nums) {
    let max = 0, maxIndex = -1;
    for(let i = 0; i < nums.length; i++) {
        if(nums[i] > max) {
            max = nums[i];
            maxIndex = i;
        }
    }
    
    for(let num of nums) {
        if(max !== num && max < (2 * num)) {
            return -1;
        }
    }
    return maxIndex;
};

File: ruby/easy/largest_number_at_least_twice_of_others.rb
# @param {Integer[]} nums
# @return {Integer}
def dominant_index(nums)
    max, max_index = 0, -1
    nums.each_with_index do |num, i|
        if num > max
            max = num
            max_index = i
        end
    end
    nums.each do |num|
        return -1 if max != num && max < (2 * num)
    end
    return max_index
end
